kyaml/yaml/util_test.go

```diff
@@ -0,0 +1,143 @@
+package yaml
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeriveSeqIndentStyle(t *testing.T) {
+	type testCase struct {
+		name           string
+		input          string
+		expectedOutput string
+	}
+
+	testCases := []testCase{
+		{
+			name: "detect simple wide indent",
+			input: `apiVersion: apps/v1
+kind: Deployment
+spec:
+  - foo
+  - bar
+  - baz
+`,
+			expectedOutput: `wide`,
+		},
+		{
+			name: "detect simple compact indent",
+			input: `apiVersion: apps/v1
+kind: Deployment
+spec:
+- foo
+- bar
+- baz
+`,
+			expectedOutput: `compact`,
+		},
+		{
+			name: "read with mixed indentation, wide first",
+			input: `apiVersion: apps/v1
+kind: Deployment
+spec:
+  - foo
+  - bar
+  - baz
+env:
+- foo
+- bar
+`,
+			expectedOutput: `wide`,
+		},
+		{
+			name: "read with mixed indentation, compact first",
+			input: `apiVersion: apps/v1
+kind: Deployment
+spec:
+- foo
+- bar
+- baz
+env:
+  - foo
+  - bar
+`,
+			expectedOutput: `compact`,
+		},
+		{
+			name: "read with mixed indentation, compact first with less elements",
+			input: `apiVersion: apps/v1
+kind: Deployment
+spec:
+- foo
+- bar
+env:
+  - foo
+  - bar
+  - baz
+`,
+			expectedOutput: `compact`,
+		},
+		{
+			name: "skip wrapped sequence strings",
+			input: `apiVersion: apps/v1
+kind: Deployment
+spec: |-
+  - foo
+  - bar
```

Comment:
Will be useful to include some additional cases for the block chomping operator https://yaml.org/spec/1.2/spec.html#id2794534

Reply:
Updated the logic and added tests for all such wrapped string cases.